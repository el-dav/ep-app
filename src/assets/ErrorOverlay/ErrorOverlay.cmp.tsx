import React, { FC } from 'react';
import { Button } from 'antd';
import { useTransition, animated } from 'react-spring';

import { styled } from 'theme';

const Container = styled(animated.div)`
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: ${({ theme }) => theme.palette.error.main};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  > * {
    margin: 20px 0px;
  }
  z-index: 10;
`;

const Message = styled.h2`
  color: ${({ theme }) => theme.palette.error.contrastText};
`;

type Props = {
  className?: string;
  errorMessage: string;
  onDissmiss: () => void;
  isShown: boolean;
};

const ErrorOverlay: FC<Props> = ({
  className,
  errorMessage,
  onDissmiss,
  isShown
}) => {
  const transitions = useTransition(isShown, null, {
    from: { opacity: 0, transform: 'translate(0, 30px)' },
    enter: { opacity: 1, transform: 'translate(0, 0)' },
    leave: { opacity: 0, transform: 'translate(0, 30px)' }
  });

  return (
    <>
      {transitions.map(
        ({ item, key, props }) =>
          item && (
            <Container key={key} className={`${className || ''}`} style={props}>
              <Message>{errorMessage}</Message>
              <Button onClick={onDissmiss}>Okay</Button>
            </Container>
          )
      )}
    </>
  );
};

export default ErrorOverlay;
