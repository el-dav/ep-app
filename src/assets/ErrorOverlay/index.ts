export { default as ErrorOverlay } from './ErrorOverlay.cmp';
