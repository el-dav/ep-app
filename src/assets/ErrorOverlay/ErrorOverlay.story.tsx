import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import { ThemeProvider, styled } from 'theme';

import ErrorOverlay from './ErrorOverlay.cmp';

const Container = styled.div`
  position: relative;
  width: 400px;
  height: 400px;
`;

storiesOf('ErrorOverlay', module).add('Default', () => {
  const errorMessage = text('Error Message', 'Order Time has Elapsed =`(');
  const isShown = boolean('Is Shown?', true);
  const onDissmiss = action('onDissmiss');

  return (
    <ThemeProvider>
      <Container>
        <ErrorOverlay
          errorMessage={errorMessage}
          onDissmiss={onDissmiss}
          isShown={isShown}
        />
      </Container>
    </ThemeProvider>
  );
});
