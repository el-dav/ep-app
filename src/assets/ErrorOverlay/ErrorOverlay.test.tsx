import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from 'theme';

import ErrorOverlay from './ErrorOverlay.cmp';

const errorMessage = 'message';
const onDismiss = () => undefined;

describe('ErrorOverlay', () => {
  it('renders the Component correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <ErrorOverlay
            isShown={false}
            errorMessage={errorMessage}
            onDissmiss={onDismiss}
          />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Component correctly when shown', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <ErrorOverlay
            isShown
            errorMessage={errorMessage}
            onDissmiss={onDismiss}
          />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
