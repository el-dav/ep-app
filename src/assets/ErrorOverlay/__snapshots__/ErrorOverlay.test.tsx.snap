// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorOverlay renders the Component correctly 1`] = `null`;

exports[`ErrorOverlay renders the Component correctly when shown 1`] = `
<div
  className=" css-idefdn"
  style={
    Object {
      "opacity": 0,
      "transform": "translate(0, 30px)",
    }
  }
>
  <h2
    className="css-n482ir"
  >
    message
  </h2>
  <button
    className="ant-btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Okay
    </span>
  </button>
</div>
`;
